src/yara_detector/yara_detector.cpp

```diff
@@ -245,11 +245,8 @@ int YaraDetector::yaraCallback(int message, void *messageData, void *userData)
  */
 bool YaraDetector::addRules(const char *string)
 {
-	auto result = yr_compiler_add_string(compiler, string, nullptr);
-	if (result == 0)
-		needsRecompilation = true;
-
-	return result == 0;
+	const auto result = yr_compiler_add_string(compiler, string, nullptr);
+	return (needsRecompilation = !(result));
```

Comment:
I would prefer readability over compactness. Even though your code got rid of the branches, when I see this I am not sure whether the author intended the assignment or he just mistyped equal to.